[Bug]: Error handling in nwb.py masks true error

### What happened?

`NWBZarrIO` is wrapped in a giant try/except that masks exceptions: https://github.com/hdmf-dev/hdmf-zarr/blob/dev/src/hdmf_zarr/nwb.py#L12

pynwb 2.8.2 is not compatible with the dev version of HDMF. When using pynwb < 2.8.3 (just released) with the dev version of HDMF (which will soon be released), and trying to import `NWBZarrIO`, you get:

```
ImportError while importing test module '/home/runner/work/hdmf-zarr/hdmf-zarr/tests/unit/test_fsspec_streaming.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/unit/test_fsspec_streaming.py:2: in <module>
    from hdmf_zarr import NWBZarrIO
/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/hdmf_zarr/__init__.py:3: in <module>
    from .nwb import NWBZarrIO
E   ImportError: cannot import name 'NWBZarrIO' from 'hdmf_zarr.nwb' (/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/hdmf_zarr/nwb.py)
```
The warning doesn't even show up for some reason, and the error hides the true error.

This shows up in the [HDMF dev compatibility tests](https://github.com/hdmf-dev/hdmf-zarr/actions/runs/11965594186/job/33359858798?pr=240) currently, until the pynwb requirement in hdmf-zarr is updated to 2.8.3.

The error should be more informative, and the import should perhaps be refactored.

### Steps to Reproduce

```shell
from .nwb import NWBZarrIO
```


### Traceback

```shell
See above
```


### Operating System

macOS

### Python Executable

Conda

### Python Version

3.12

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/hdmf-dev/hdmf-zarr/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@
 
 ### Bug Fixes
 * Fix reading of cached specs and caching of specs during export. @rly [#232](https://github.com/hdmf-dev/hdmf-zarr/pull/232)
+* Fix hiding of pynwb compatibility errors. @rly [242](https://github.com/hdmf-dev/hdmf-zarr/pull/242)
 
 ## 0.9.0 (September 16, 2024)
 ### Enhancements

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -123,7 +123,7 @@ exclude = [
 ]
 line-length = 120
 
-[tool.ruff.per-file-ignores]
+[tool.ruff.lint.per-file-ignores]
 "docs/gallery/*" = ["E402", "T201"]
 "src/*/__init__.py" = ["F401"]
 "test_gallery.py" = ["T201"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,11 +29,11 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Medical Science Apps."
 ]
 dependencies = [
-    'hdmf>=3.14.3',
+    'hdmf>=3.14.5',
     'zarr>=2.11.0, <3.0', # pin below 3.0 until HDMF-zarr supports zarr 3.0
     'numpy>=1.24',
     'numcodecs>=0.9.1',
-    'pynwb>=2.5.0',
+    'pynwb>=2.8.3',
     'threadpoolctl>=3.1.0',
 ]
 dynamic = ["version"]

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,7 +2,8 @@
 hdmf==3.14.5
 zarr==2.18.3; python_version >= "3.10"  # zarr 2.18.3 dropped support for python 3.9
 zarr==2.18.2; python_version < "3.10"
-pynwb==2.8.2
-numpy==2.1.3
+pynwb==2.8.3
+numpy==2.1.3; python_version >= "3.10"  # numpy 2.1.0 dropped support for python 3.9
+numpy==2.0.2; python_version < "3.10"
 numcodecs==0.13.1
 threadpoolctl==3.5.0

--- a/src/hdmf_zarr/nwb.py
+++ b/src/hdmf_zarr/nwb.py
@@ -9,85 +9,82 @@ from hdmf.utils import (docval,
 from hdmf.backends.io import HDMFIO
 from hdmf.build import (BuildManager,
                         TypeMap)
-try:
-    from pynwb import get_manager, get_type_map
+from pynwb import get_manager, get_type_map
 
-    class NWBZarrIO(ZarrIO):
+
+class NWBZarrIO(ZarrIO):
+    """
+    IO backend for PyNWB for writing NWB files
+
+    This class is similar to the :py:class:`~pynwb.NWBHDF5IO` class in PyNWB. The main purpose of this class
+    is to perform default setup for BuildManager, loading or namespaces etc., in the context
+    of the NWB format.
+    """
+    @docval(*get_docval(ZarrIO.__init__),
+            {'name': 'load_namespaces', 'type': bool,
+                'doc': 'whether or not to load cached namespaces from given path - not applicable in write mode',
+                'default': True},
+            {'name': 'extensions', 'type': (str, TypeMap, list),
+                'doc': 'a path to a namespace, a TypeMap, or a list consisting paths  to namespaces and TypeMaps',
+                'default': None})
+    def __init__(self, **kwargs):
+        path, mode, manager, extensions, load_namespaces, synchronizer, storage_options = \
+            popargs('path', 'mode', 'manager', 'extensions',
+                    'load_namespaces', 'synchronizer', 'storage_options', kwargs)
+
+        io_modes_that_create_file = ['w', 'w-', 'x']
+        if mode in io_modes_that_create_file or manager is not None or extensions is not None:
+            load_namespaces = False
+
+        if load_namespaces:
+            tm = get_type_map()
+            super(NWBZarrIO, self).load_namespaces(tm, path, storage_options)
+            manager = BuildManager(tm)
+        else:
+            if manager is not None and extensions is not None:
+                raise ValueError("'manager' and 'extensions' cannot be specified together")
+            elif extensions is not None:
+                manager = get_manager(extensions=extensions)
+            elif manager is None:
+                manager = get_manager()
+        super(NWBZarrIO, self).__init__(path,
+                                        manager=manager,
+                                        mode=mode,
+                                        synchronizer=synchronizer,
+                                        storage_options=storage_options)
+
+    @docval({'name': 'src_io', 'type': HDMFIO, 'doc': 'the HDMFIO object for reading the data to export'},
+            {'name': 'nwbfile', 'type': 'NWBFile',
+                'doc': 'the NWBFile object to export. If None, then the entire contents of src_io will be exported',
+                'default': None},
+            {'name': 'write_args', 'type': dict, 'doc': 'arguments to pass to :py:meth:`write_builder`',
+                'default': dict()})
+    def export(self, **kwargs):
+        nwbfile = popargs('nwbfile', kwargs)
+        kwargs['container'] = nwbfile
+        super().export(**kwargs)
+
+    @staticmethod
+    @docval({'name': 'path',
+                'type': (str, Path, *SUPPORTED_ZARR_STORES),
+                'doc': 'the path to the Zarr file or a supported Zarr store'},
+            is_method=False)
+    def read_nwb(**kwargs):
         """
-        IO backend for PyNWB for writing NWB files
-
-        This class is similar to the :py:class:`~pynwb.NWBHDF5IO` class in PyNWB. The main purpose of this class
-        is to perform default setup for BuildManager, loading or namespaces etc., in the context
-        of the NWB format.
+        Helper factory method for reading an NWB file and return the NWBFile object
         """
-        @docval(*get_docval(ZarrIO.__init__),
-                {'name': 'load_namespaces', 'type': bool,
-                 'doc': 'whether or not to load cached namespaces from given path - not applicable in write mode',
-                 'default': True},
-                {'name': 'extensions', 'type': (str, TypeMap, list),
-                 'doc': 'a path to a namespace, a TypeMap, or a list consisting paths  to namespaces and TypeMaps',
-                 'default': None})
-        def __init__(self, **kwargs):
-            path, mode, manager, extensions, load_namespaces, synchronizer, storage_options = \
-                popargs('path', 'mode', 'manager', 'extensions',
-                        'load_namespaces', 'synchronizer', 'storage_options', kwargs)
+        # Retrieve the filepath
+        path = popargs('path', kwargs)
+        if isinstance(path, Path):
+            path = str(path)
+        # determine default storage options to use when opening a file from S3
+        storage_options = {}
+        if isinstance(path, str) and path.startswith(("s3://")):
+            storage_options = dict(anon=True)
 
-            io_modes_that_create_file = ['w', 'w-', 'x']
-            if mode in io_modes_that_create_file or manager is not None or extensions is not None:
-                load_namespaces = False
+        # open the file with NWBZarrIO and rad the file
+        io = NWBZarrIO(path=path, mode="r", load_namespaces=True, storage_options=storage_options)
+        nwbfile = io.read()
 
-            if load_namespaces:
-                tm = get_type_map()
-                super(NWBZarrIO, self).load_namespaces(tm, path, storage_options)
-                manager = BuildManager(tm)
-            else:
-                if manager is not None and extensions is not None:
-                    raise ValueError("'manager' and 'extensions' cannot be specified together")
-                elif extensions is not None:
-                    manager = get_manager(extensions=extensions)
-                elif manager is None:
-                    manager = get_manager()
-            super(NWBZarrIO, self).__init__(path,
-                                            manager=manager,
-                                            mode=mode,
-                                            synchronizer=synchronizer,
-                                            storage_options=storage_options)
-
-        @docval({'name': 'src_io', 'type': HDMFIO, 'doc': 'the HDMFIO object for reading the data to export'},
-                {'name': 'nwbfile', 'type': 'NWBFile',
-                 'doc': 'the NWBFile object to export. If None, then the entire contents of src_io will be exported',
-                 'default': None},
-                {'name': 'write_args', 'type': dict, 'doc': 'arguments to pass to :py:meth:`write_builder`',
-                 'default': dict()})
-        def export(self, **kwargs):
-            nwbfile = popargs('nwbfile', kwargs)
-            kwargs['container'] = nwbfile
-            super().export(**kwargs)
-
-        @staticmethod
-        @docval({'name': 'path',
-                 'type': (str, Path, *SUPPORTED_ZARR_STORES),
-                 'doc': 'the path to the Zarr file or a supported Zarr store'},
-                is_method=False)
-        def read_nwb(**kwargs):
-            """
-            Helper factory method for reading an NWB file and return the NWBFile object
-            """
-            # Retrieve the filepath
-            path = popargs('path', kwargs)
-            if isinstance(path, Path):
-                path = str(path)
-            # determine default storage options to use when opening a file from S3
-            storage_options = {}
-            if isinstance(path, str) and path.startswith(("s3://")):
-                storage_options = dict(anon=True)
-
-            # open the file with NWBZarrIO and rad the file
-            io = NWBZarrIO(path=path, mode="r", load_namespaces=True, storage_options=storage_options)
-            nwbfile = io.read()
-
-            # return the NWBFile object
-            return nwbfile
-
-except ImportError:
-    warn("PyNWB is not installed. Support for NWBZarrIO is disabled.")
+        # return the NWBFile object
+        return nwbfile

--- a/src/hdmf_zarr/nwb.py
+++ b/src/hdmf_zarr/nwb.py
@@ -1,5 +1,4 @@
 """Module with Zarr backend for NWB for integration with PyNWB"""
-from warnings import warn
 from pathlib import Path
 from .backend import ZarrIO, SUPPORTED_ZARR_STORES
 

